mne/forward/_field_interpolation.py

```diff
@@ -87,6 +91,127 @@ def _compute_mapping_matrix(fmd, info):
     return mapping_mat
 
 
+def compute_virtual_evoked(evoked, from_type='mag', to_type='grad',
+                           copy=True, baseline=None, n_jobs=1):
+    """Compute virtual evoked using interpolated fields in mag/grad channels.
+
+    Parameters
+    ----------
+    evoked : instance of mne.Evoked
+        The evoked object.
+    from_type : str
+        The original channel type. It can be 'mag' or 'grad'.
+    to_type : str
+        The destination channel type. It can be 'mag' or 'grad'.
+    copy : bool
+        If False evoked is modified in place.
+    n_jobs : int
+        Number of jobs to run in parallel.
+
+    Returns
+    -------
+    evoked : instance of mne.Evoked
+        The transformed evoked object containing only virtual channels.
+    """
+    if from_type not in ['mag', 'grad']:
+        raise ValueError('from_type must be "mag" or "grad", not "%s"'
+                         % from_type)
+    if to_type not in ['mag', 'grad']:
+        raise ValueError('to_type must be "mag" or "grad", not "%s"'
+                         % to_type)
+    if from_type == to_type:
+        raise ValueError('from_type and to_type cannot be the same.')
+
+    # get surface in head coordinates
+    surf_virt = get_meg_helmet_surf(evoked.info)
+
+    # pick the original and destination channels
+    pick_from = pick_types(evoked.info, meg=from_type, eeg=False,
+                           ref_meg=False)
+    pick_to = pick_types(evoked.info, meg=to_type, eeg=False,
```

Comment:
On Fri, Apr 17, 2015 at 10:07 PM, Eric Larson <notifications@github.com>
wrote:

> In mne/forward/_field_interpolation.py
> <https://github.com/mne-tools/mne-python/pull/1943#discussion_r28625950>:
>
> > +    if from_type not in ['mag', 'grad']:
> > +        raise ValueError('from_type must be "mag" or "grad", not "%s"'
> > +                         % from_type)
> > +    if to_type not in ['mag', 'grad']:
> > +        raise ValueError('to_type must be "mag" or "grad", not "%s"'
> > +                         % to_type)
> > +    if from_type == to_type:
> > +        raise ValueError('from_type and to_type cannot be the same.')
> > +
> > +    # get surface in head coordinates
> > +    surf_virt = get_meg_helmet_surf(evoked.info)
> > +
> > +    # pick the original and destination channels
> > +    pick_from = pick_types(evoked.info, meg=from_type, eeg=False,
> > +                           ref_meg=False)
> > +    pick_to = pick_types(evoked.info, meg=to_type, eeg=False,
>
> Exactly -- another particular use case of the general idea I mentioned
> above. I think that should also be part of the release, even if it means
> delaying a week or two to get it in. @mainakjas
> <https://github.com/mainakjas> do you have the bandwidth to really work
> on this in the coming week so we can get the generalization working, and
> then the particular use cases? I can see three: virtual channels (done),
> proper interpolation, and mapping data between head positions. The last two
> should be fairly straightforward if the generalization is good.
>
cc @yousrabk who will give the third application a shot :)


> My vote is to try to get all three done, and if it ends up being too
> ambitious, we can scale back later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/1943/files#r28625950>.
>